src/ScriptEditor/ui/ScriptEditorRoot.tsx

```diff
@@ -605,16 +631,26 @@ export function Root(props: IProps): React.ReactElement {
     openScripts = items;
   }
 
+  function currentTabIndex(): number | undefined {
+    if (currentScript !== null) {
+      return openScripts.findIndex(
+          (script) =>
+            currentScript !== null &&
+            script.fileName === currentScript.fileName &&
+            script.hostname === currentScript.hostname,
+        );
+    } 
+
+    return undefined;
+  }
+
   function onTabClick(index: number): void {
     if (currentScript !== null) {
       // Save currentScript to openScripts
-      const curIndex = openScripts.findIndex(
-        (script) =>
-          currentScript !== null &&
-          script.fileName === currentScript.fileName &&
-          script.hostname === currentScript.hostname,
-      );
-      openScripts[curIndex] = currentScript;
+      const curIndex = currentTabIndex();
+      if (curIndex !== undefined) {
```

Comment:
maybe throw an error if the tab click didnt work for whatever reason?
`dialogBoxCreate("ERRORTHING");`

Reply:
let me look into this (although, I believe the logic hasn't changed, as undefined is only returned when currentScript === null, which is the same check this function did before calling it; unless there is some other async-ness involved on modifying currentScript)

Reply:
yeah i dont think it was doing it before, but i felt it might aswell tell the user something went wrong. feel free to disregard :)